site/_posts/2020-11-15-how-to-securely-implement-oauth-rails.md

```diff
@@ -0,0 +1,300 @@
+
+Once your Docker container is up, it should be running at [localhost:9011](http://localhost:9011). Navigate there and finish the setup steps. 
+When those are completed, login as the administrator that you just created. Now we are ready to rock!
+
+First things first. A [FusionAuth Application](https://fusionauth.io/docs/v1/tech/core-concepts/applications/) is simply something a user can log into.
+As such, we want to create a new application for our Rails app.
+
+### Create an Application
+Select `Applications` from the left nav bar. Then click the green plus sign in the upper-right hand corner to add a new application.
+We will want to configure a few pieces of important information:
+* **Name** (Required): A display name for our application. Enter `fusionauth-rails-app`.
+* **Authorized redirect URL's**: Found under the `OAuth` tab. This URL tells FusionAuth where to redirect to after a user successfully authenticates. 
+* **Logout URL**: Also found under the `OAuth` tab. This URL tells FusionAuth where to redirect to when a user logs out.
+
+{% include _image.liquid src="/assets/img/blogs/fusionauth-example-rails/add-application.png" alt="Creating a new Application in FusionAuth." class="img-fluid" figure=false %}
+
+### Create a User
+For this example, we will manually create a user. Click on `Users` in the left-nav bar and then click the green plus sign in the upper-right hand corner to add a new user.
+
+{% include _image.liquid src="/assets/img/blogs/fusionauth-example-rails/create-user.png" alt="Creating a new User in FusionAuth." class="img-fluid" figure=false %}
+
+Configure the following attributes and then save the new user:
+* **Email**
+* **Password and Confirm**: Toggle `Send email to setup password` to off and manually enter a password for the user.
+
+### Register the User
+Looking good! Now we have our new application and a user. The last thing we want to do is `register` the user with our `application`.
+Again, navigate to the `Users` page, find our newly created user, and click the `Manage` button.
+
+Click on the `Add Registration` button, select the Rails application we created, and save.
+
+{% include _image.liquid src="/assets/img/blogs/fusionauth-example-rails/user-registered.png" alt="The Manage User page in FusionAuth." class="img-fluid" figure=false %}
+
+## Build the Rails Application
+Lets get started building a very basic Rails application to demonstrate authentication with FusionAuth.
+
+Create a new Rails application.
+```bash
+rails new rails-fusionauth-app
+```
+
+Our `Gemfile` will use a few additional dependencies so go ahead and add them before we get started.
+```ruby
+gem 'jwt'
+gem 'oauth'
+```
+
+Install dependencies
+```bash
+bundle install
+```
+
+### Configuration
+Before we go any further, there are some critical `Application` attributes we will need from our FusionAuth configuration so that our Rails app knows how to successfully communicate with it.
+Specifically:
+* `client_id`
```

Comment:
Isn't this called "Client Id" in the UI? or are you pulling it from somewhere else? Best to have this match up with the UI if possible. Same for all these other configuration params.